src/python-common/ceph/deployment/service_spec.py

```diff
@@ -778,3 +780,121 @@ def __init__(self,
 
 
 yaml.add_representer(AlertManagerSpec, ServiceSpec.yaml_representer)
+
+
+class CustomContainerSpec(ServiceSpec):
+    props: List[str] = [
+        'image', 'entrypoint', 'uid', 'gid', 'args', 'volume_mounts',
+        'container_args', 'envs', 'privileged', 'ptrace', 'bind_mounts',
+        'ports', 'dirs', 'files'
+    ]
+
+    def __init__(self,
+                 service_type: str = 'container',
+                 service_id: str = None,
+                 placement: Optional[PlacementSpec] = None,
+                 unmanaged: bool = False,
+                 preview_only: bool = False,
+                 image: str = None,
+                 entrypoint: Optional[str] = None,
+                 uid: Optional[int] = None,
+                 gid: Optional[int] = None,
+                 args: Optional[List[str]] = [],
+                 volume_mounts: Optional[Dict[str, str]] = {},
+                 container_args: Optional[List[str]] = [],
+                 envs: Optional[List[str]] = [],
+                 privileged: Optional[bool] = False,
+                 bind_mounts: Optional[List[List[str]]] = None,
+                 ptrace: Optional[bool] = False,
```

Comment:
can you leave this out please? we already have

https://github.com/ceph/ceph/blob/a8bf3e9b2f804d11600d412dae0b9493a3df61c8/src/pybind/mgr/cephadm/module.py#L200